Added feature for listing holidays for the running month (statewise for India)

siddhant-nair opened the pull request:
- Added Holiday feature.
- You can check holidays in your state for the current month 
- Data is fetched from a local json created for 2025

![Screenshot from 2025-01-07 12-21-16](https://github.com/user-attachments/assets/6a490669-be3a-415d-a826-384b570e4b37)

knadh commented:
This is nice. Let's add country-code suffix support and make it univeral. Eg:  kerala.holidays.in or holidays.us etc.

- Query comes in as *.holiday.$code or .holidays.$code
- Refactor the internal map to [$code][state]
- Files in the data dir can be holidays.$code.json

siddhant-nair commented:
Was adding country support and figured I might as well add TLD support in general

- Added a new function in handlers called registerWithTLDSupport
- Added holidays of 10 countries!
- if state name is not mentioned only the national holidays are displayed

knadh commented:
This is great! If you can fix the merge conflict, we can merge and take this live.

siddhant-nair commented:
All done!

You might want to check the config once though. 
The nanoid limits in the main.go file (nanoid.max_results & nanoid.max_length) are missing

knadh commented:
Cheers. Will test and merge it this week.
